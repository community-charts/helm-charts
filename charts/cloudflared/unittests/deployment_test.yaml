suite: test deployment

templates:
  - deployment.yaml
  - configmap.yaml

tests:
  - it: should be custom service account when we do not create it
    set:
      serviceAccount:
        create: false
        name: customsa
      tunnelConfig:
        name: unittest
    asserts:
      - equal:
          path: spec.template.spec.serviceAccountName
          value: customsa
        template: deployment.yaml

  - it: should use DaemonSet when all nodes flag enabled
    set:
      replica:
        allNodes: true
      tunnelConfig:
        name: unittest
    asserts:
      - isKind:
          of: DaemonSet
        template: deployment.yaml

  - it: should use Deployment when all nodes flag disabled
    set:
      replica:
        allNodes: false
      tunnelConfig:
        name: unittest
    asserts:
      - isKind:
          of: Deployment
        template: deployment.yaml

  - it: should use update strategy when all nodes flag enabled
    set:
      replica:
        allNodes: true
      updateStrategy:
        type: RollingUpdate
        rollingUpdate:
          maxUnavailable: 1
      tunnelConfig:
        name: unittest
    asserts:
      - equal:
          path: spec.updateStrategy.type
          value: RollingUpdate
        template: deployment.yaml
      - equal:
          path: spec.updateStrategy.rollingUpdate.maxUnavailable
          value: 1
        template: deployment.yaml

  - it: should use replica when all nodes flag disabled
    set:
      replica:
        allNodes: false
        count: 5
      tunnelConfig:
        name: unittest
    asserts:
      - equal:
          path: spec.replicas
          value: 5
        template: deployment.yaml

  - it: should set termination grace period
    set:
      terminationGracePeriodSeconds: 10
      tunnelConfig:
        name: unittest
    asserts:
      - equal:
          path: spec.template.spec.terminationGracePeriodSeconds
          value: 10
        template: deployment.yaml

  - it: should set credentials from tunnel-credentials when base64EncodedPemFile and base64EncodedConfigJsonFile fields set
    set:
      tunnelConfig:
        name: unittest
      tunnelSecrets:
        base64EncodedPemFile: UEVNLUZJTEUtQ09OVEVOVA==
        base64EncodedConfigJsonFile: eyJmaWxlIjoiQ29uZmlnSnNvbkZpbGUifQ==
    asserts:
      - contains:
          path: spec.template.spec.volumes
          content:
            name: creds
            projected:
              sources:
                - secret:
                    name: tunnel-credentials
        template: deployment.yaml

  - it: should set pem file from existing secret when existingPemFileSecret name set
    set:
      tunnelConfig:
        name: unittest
      tunnelSecrets:
        base64EncodedConfigJsonFile: eyJmaWxlIjoiQ29uZmlnSnNvbkZpbGUifQ==
        existingPemFileSecret:
          name: my-pem-file-secret
    asserts:
      - contains:
          path: spec.template.spec.volumes
          content:
            name: creds
            projected:
              sources:
                - secret:
                    name: tunnel-credentials
                - secret:
                    name: my-pem-file-secret
        template: deployment.yaml

  - it: should set config json file from existing secret when existingConfigJsonFileSecret name set
    set:
      tunnelConfig:
        name: unittest
      tunnelSecrets:
        base64EncodedPemFile: UEVNLUZJTEUtQ09OVEVOVA==
        existingConfigJsonFileSecret:
          name: my-config-json-file-secret
    asserts:
      - contains:
          path: spec.template.spec.volumes
          content:
            name: creds
            projected:
              sources:
                - secret:
                    name: tunnel-credentials
                - secret:
                    name: my-config-json-file-secret
        template: deployment.yaml

  - it: should set pem file and config json file from existing secrets when existingPemFileSecret name and existingConfigJsonFileSecret name set
    set:
      tunnelConfig:
        name: unittest
      tunnelSecrets:
        existingPemFileSecret:
          name: my-pem-file-secret
        existingConfigJsonFileSecret:
          name: my-config-json-file-secret
    asserts:
      - contains:
          path: spec.template.spec.volumes
          content:
            name: creds
            projected:
              sources:
                - secret:
                    name: my-pem-file-secret
                - secret:
                    name: my-config-json-file-secret
        template: deployment.yaml

  - it: should set TUNNEL_ORIGIN_CERT environment variable
    set:
      tunnelConfig:
        name: unittest
      tunnelSecrets:
        existingPemFileSecret:
          name: my-pem-file-secret
          key: custom.pem
        existingConfigJsonFileSecret:
          name: my-config-json-file-secret
    asserts:
      - contains:
          path: spec.template.spec.containers[?(@.name == "cloudflared")].env
          content:
            name: TUNNEL_ORIGIN_CERT
            value: /etc/cloudflared/creds/custom.pem
        template: deployment.yaml

  - it: should match snapshot of default values
    set:
      tunnelConfig:
        name: unittest
    release:
      name: cloudflared
      namespace: cloudflare
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
