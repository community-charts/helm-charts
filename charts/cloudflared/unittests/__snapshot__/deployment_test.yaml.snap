should match snapshot of default values:
  1: |
    apiVersion: v1
    data:
      config.yaml: "\ntunnel: unittest\ncredentials-file: /etc/cloudflared/creds/credentials.json\noriginRequest:\n  connectTimeout: 30s\n\ningress: \n  - hostname: example.com\n    service: http://traefik.kube-system.svc.cluster.local:80\n  - service: http_status:404\n\nmetrics: 0.0.0.0:2000\nmetrics-update-freq: 5s\n\nautoupdate-freq: 24h\nno-autoupdate: true\n\ngrace-period: 30s\n\nretries: 5\n\n# auto, http2, h2mux, quic\nprotocol: auto\n\n# info, warn, error, fatal, panic\nloglevel: info\ntransport-loglevel: warn\n"
    kind: ConfigMap
    metadata:
      name: cloudflared
  2: |
    apiVersion: apps/v1
    kind: DaemonSet
    metadata:
      labels:
        app: cloudflared
        app.kubernetes.io/instance: cloudflared
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: cloudflared
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: cloudflared-1.0.0
      name: cloudflared
    spec:
      selector:
        matchLabels:
          app: cloudflared
          app.kubernetes.io/instance: cloudflared
          app.kubernetes.io/name: cloudflared
      template:
        metadata:
          annotations:
            checksum/config: 6914168e330d7d7832f59de17cb31e7f60be876dd10f1e52413ad6057a1f8076
          labels:
            app: cloudflared
            app.kubernetes.io/instance: cloudflared
            app.kubernetes.io/name: cloudflared
        spec:
          containers:
            - args:
                - tunnel
                - --config
                - /etc/cloudflared/config/config.yaml
                - run
              env:
                - name: TUNNEL_ORIGIN_CERT
                  value: /etc/cloudflared/creds/cert.pem
              image: cloudflare/cloudflared:1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /ready
                  port: 2000
                initialDelaySeconds: 10
                periodSeconds: 10
              name: cloudflared
              ports:
                - containerPort: 2000
                  name: active-con-stat
                  protocol: TCP
              resources: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  add: []
                  drop:
                    - ALL
                privileged: false
                readOnlyRootFilesystem: true
                runAsGroup: 65532
                runAsNonRoot: true
                runAsUser: 65532
              volumeMounts:
                - mountPath: /etc/cloudflared/config
                  name: config
                  readOnly: true
                - mountPath: /etc/cloudflared/creds
                  name: creds
                  readOnly: true
          nodeSelector:
            kubernetes.io/os: linux
          securityContext:
            fsGroup: 65532
            fsGroupChangePolicy: OnRootMismatch
            sysctls:
              - name: net.ipv4.ping_group_range
                value: 0 2147483647
          serviceAccountName: cloudflared
          terminationGracePeriodSeconds: 30
          tolerations:
            - effect: NoSchedule
              operator: Exists
          volumes:
            - name: creds
              projected:
                sources:
                  - secret:
                      name: tunnel-credentials
            - configMap:
                items:
                  - key: config.yaml
                    path: config.yaml
                name: cloudflared
              name: config
      updateStrategy:
        rollingUpdate:
          maxUnavailable: 1
        type: RollingUpdate
